src/api/api.ts

```diff
@@ -1,25 +1,271 @@
-import axios, { AxiosError } from "axios";
-import { APIError, APIResponse, APIResponseFailure } from "../shared/types";
-import ERRORS from "../shared/errors";
+import axios, { AxiosRequestConfig, AxiosResponse } from "axios";
+import {
+  APIResponse,
+  APIError,
+  Issue,
+  Article,
+  User,
+  UserAPIError,
+  IssueAPIError,
+  Pagination,
+  ArticleContent,
+} from "../shared/types";
+import { ProfileFormVals } from "../profile/ProfileForm";
 
 export const API_ROOT = "/api/";
 
-export const getApiUrl = (url: string) => {
-  if (!url) return API_ROOT;
-  return `${API_ROOT}${url.split("/").filter(p => !!p).join("/")}/`;
+type RequestConfig = Omit<AxiosRequestConfig, "baseURL" | "validateStatus">;
+
+const axiosRequest = async (config: RequestConfig) => {
+  const baseConfig: AxiosRequestConfig = {
+    baseURL: API_ROOT,
+    validateStatus: () => true,
+  };
+  return await axios({ ...baseConfig, ...config });
```

Comment:
So, I think we've been going back and forth on whether the status code should be in all responses, or on just errors. I would prefer having them on all responses since it doesn't really harm anything, and having `status_code` on `APIErrors` makes them hard to deal with because we have to special case exclude the field. Thoughts?

Reply:
Would agree for all responses. So are you suggesting something like the following?

```js
type APIResponse<S, F> = {
  status_code: number;
} &(S | F);
```

Reply:
Yeah, pretty much.

Unrelated question: why do we have the last two type parameters here?
```
export type APIResponse<
  T,
  U extends APIError = APIError,
  S extends APIResponseSuccess<T> = APIResponseSuccess<T>,
  F extends APIResponseFailure<U> = APIResponseFailure<U>
> = { status_code: number } & (S | F);
```

I don't think anything is going to extend `APIResponseSuccess` or `APIResponseFailure`, and they make the Intellisense type hints obnoxiously long.

Reply:
https://github.com/UWmathNEWS/slugline-web/pull/33/commits/34cb54d1b3c67ab3bd8d64f5ae70e3b8b20d61dc

Reply:
I ended up getting rid of the type parameters for now, I'll leave this open in case you have any objections.

Reply:
Their only purpose, if I recall correctly, was to allow for UserAPIResponse (which was typed as `APIResponse<User, UserAPIError, UserAPIResponseSuccess>` or something like that). Seeing as we've removed that distinction, I'm ok with removing those type parameters.